app/src/main/java/org/p2p/wallet/striga/user/repository/StrigaUserRepositoryMapper.kt

```diff
@@ -28,6 +35,60 @@ class StrigaUserRepositoryMapper {
         }
     }
 
+    fun fromNetwork(response: StrigaCreateUserResponse): StrigaUserInitialDetails {
+        return try {
+            StrigaUserInitialDetails(
+                userId = response.userId,
+                email = response.email,
+                kycStatus = response.toKycDetails()
+            )
+        } catch (mappingFailed: Throwable) {
+            throw StrigaDataLayerError.InternalError(mappingFailed, "StrigaUserInitialDetails mapping failed")
+        }
+    }
+
+    fun toNetwork(signupData: List<StrigaSignupData>): StrigaCreateUserRequest {
+        val map = signupData.filter { it.value != null }.associate { it.type to it.value!! }
+
+        val birthday = map.getOrThrow(StrigaSignupDataType.DATE_OF_BIRTH).split(".")
+
+        return StrigaCreateUserRequest(
+            firstName = map.getOrThrow(StrigaSignupDataType.FIRST_NAME),
+            lastName = map.getOrThrow(StrigaSignupDataType.LAST_NAME),
+            userEmail = map.getOrThrow(StrigaSignupDataType.EMAIL),
+            // TODO: documentation is empty in this case
+            documentIssuingCountry = map.getOrThrow(StrigaSignupDataType.COUNTRY_OF_BIRTH),
+            // TODO: the same
+            nationality = map.getOrThrow(StrigaSignupDataType.COUNTRY_OF_BIRTH),
+            mobilePhoneDetails = StrigaCreateUserRequest.MobileRequest(
+                number = map.getOrThrow(StrigaSignupDataType.PHONE_NUMBER),
+                countryCode = map.getOrThrow(StrigaSignupDataType.PHONE_NUMBER)
+            ),
+            dateOfBirth = StrigaCreateUserRequest.DateOfBirthRequest(
+                day = birthday[0],
+                month = birthday[1],
+                year = birthday[2]
+            ),
+            address = StrigaCreateUserRequest.AddressRequest(
+                addressLine1 = map.getOrThrow(StrigaSignupDataType.CITY_ADDRESS_LINE),
+                addressLine2 = "",
+                city = map.getOrThrow(StrigaSignupDataType.CITY),
+                state = map.getOrThrow(StrigaSignupDataType.CITY_STATE),
+                country = map.getOrThrow(StrigaSignupDataType.COUNTRY),
+                postalCode = map.getOrThrow(StrigaSignupDataType.CITY_POSTAL_CODE)
+            ),
+            occupation = map.getOrThrow(StrigaSignupDataType.OCCUPATION),
+            sourceOfFunds = map.getOrThrow(StrigaSignupDataType.SOURCE_OF_FUNDS),
+            // TODO: it doesn't make sense to get it here, we need some external service
+            ipAddress = "",
```

Comment:
emptyString()

Reply:
I've removed this field, as it is not required